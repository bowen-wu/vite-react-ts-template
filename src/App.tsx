import React, { useReducer } from 'react';
import scopedClasses from './utils/scopedClasses';
import Layout from './layouts/layout';
import Routers from './routers';
import store from './stroes/store';
import Context from './stroes/context';
import { storeReducer } from './stroes/reducers';

import './App.scss';

const sc = scopedClasses('App');

const App = () => {
    const [ state, dispatch ] = useReducer(storeReducer, store);
    return (
        <div className={sc()}>
            <Context.Provider value={[state, dispatch]}>
                <Layout>
                    <Routers/>
                </Layout>
            </Context.Provider>
        </div>
    );
};

export default App;
