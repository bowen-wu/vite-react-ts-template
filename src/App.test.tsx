import * as React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from './App';

describe('Test App router', () => {
  afterEach(cleanup);

  it('take a snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('full app rendering/navigating', () => {
    const { container } = render(<App />);
    expect(container.querySelector('div.App')).toBeTruthy();
    expect(container.querySelector('div.layout')).toBeTruthy();
    expect(container.querySelector('div.home')).toBeTruthy();
  });
});
