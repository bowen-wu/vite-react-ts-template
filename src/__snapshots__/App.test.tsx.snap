// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App router take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="layout"
    >
      <div
        class="ant-design-pro ant-pro-basicLayout screen-md ant-pro-basicLayout-side"
      >
        <section
          class="ant-layout ant-layout-has-sider"
          style="min-height: 100%;"
        >
          <aside
            class="ant-layout-sider ant-layout-sider-dark ant-pro-sider ant-pro-sider-layout-side"
            style="overflow: hidden; flex: 0 0 208px; max-width: 208px; min-width: 208px; width: 208px;"
          >
            <div
              class="ant-layout-sider-children"
            >
              <div
                class="ant-pro-sider-logo"
                id="logo"
              >
                <div
                  style="color: rgb(255, 255, 255);"
                >
                  左上角标题
                </div>
              </div>
              <div
                style="flex: 1; overflow-y: auto; overflow-x: hidden;"
              >
                <ul
                  class="ant-menu ant-menu-root ant-menu-inline ant-menu-dark ant-pro-sider-menu"
                  data-menu-list="true"
                  role="menu"
                  style="width: 100%;"
                  tabindex="0"
                >
                  <li
                    class="ant-menu-item ant-menu-item-only-child"
                    data-menu-id="rc-menu-uuid-test-/welcome"
                    role="menuitem"
                    style="padding-left: 16px;"
                    tabindex="-1"
                  >
                    <span
                      class="ant-menu-title-content"
                    >
                      <a>
                        <span
                          class="ant-pro-menu-item"
                        >
                          <span
                            class="ant-pro-menu-item-title"
                          >
                            欢迎
                          </span>
                        </span>
                      </a>
                    </span>
                  </li>
                  <li
                    class="ant-menu-submenu ant-menu-submenu-inline"
                    role="none"
                  >
                    <div
                      aria-controls="rc-menu-uuid-test-/test-popup"
                      aria-expanded="false"
                      aria-haspopup="true"
                      class="ant-menu-submenu-title"
                      data-menu-id="rc-menu-uuid-test-/test"
                      role="menuitem"
                      style="padding-left: 16px;"
                      tabindex="-1"
                    >
                      <span
                        class="ant-menu-title-content"
                      >
                        <span
                          class="ant-pro-menu-item"
                          title="测试"
                        >
                          测试
                        </span>
                      </span>
                      <i
                        class="ant-menu-submenu-arrow"
                      />
                    </div>
                  </li>
                  <li
                    class="ant-menu-submenu ant-menu-submenu-inline"
                    role="none"
                  >
                    <div
                      aria-controls="rc-menu-uuid-test-/thisistest-popup"
                      aria-expanded="false"
                      aria-haspopup="true"
                      class="ant-menu-submenu-title"
                      data-menu-id="rc-menu-uuid-test-/thisistest"
                      role="menuitem"
                      style="padding-left: 16px;"
                      tabindex="-1"
                    >
                      <span
                        class="ant-menu-title-content"
                      >
                        <span
                          class="ant-pro-menu-item"
                          title="管理页"
                        >
                          管理页
                        </span>
                      </span>
                      <i
                        class="ant-menu-submenu-arrow"
                      />
                    </div>
                  </li>
                  <li
                    class="ant-menu-submenu ant-menu-submenu-inline"
                    role="none"
                  >
                    <div
                      aria-controls="rc-menu-uuid-test-/list-popup"
                      aria-expanded="false"
                      aria-haspopup="true"
                      class="ant-menu-submenu-title"
                      data-menu-id="rc-menu-uuid-test-/list"
                      role="menuitem"
                      style="padding-left: 16px;"
                      tabindex="-1"
                    >
                      <span
                        class="ant-menu-title-content"
                      >
                        <span
                          class="ant-pro-menu-item"
                          title="列表页"
                        >
                          列表页
                        </span>
                      </span>
                      <i
                        class="ant-menu-submenu-arrow"
                      />
                    </div>
                  </li>
                </ul>
                <div
                  aria-hidden="true"
                  style="display: none;"
                />
              </div>
              <div
                class="ant-pro-sider-links"
              >
                <ul
                  class="ant-menu ant-menu-root ant-menu-inline ant-menu-dark ant-pro-sider-link-menu"
                  data-menu-list="true"
                  role="menu"
                  tabindex="0"
                >
                  <li
                    class="ant-menu-item ant-menu-item-only-child ant-pro-sider-collapsed-button"
                    data-menu-id="rc-menu-uuid-test-collapsed"
                    role="menuitem"
                    style="padding-left: 16px;"
                    tabindex="-1"
                  >
                    <span
                      class="ant-menu-title-content"
                    >
                      <span
                        aria-label="menu-fold"
                        class="anticon anticon-menu-fold"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="menu-fold"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
                          />
                        </svg>
                      </span>
                    </span>
                  </li>
                </ul>
                <div
                  aria-hidden="true"
                  style="display: none;"
                />
              </div>
            </div>
          </aside>
          <div
            class="ant-layout"
            style="position: relative;"
          >
            <header
              class="ant-layout-header ant-pro-fixed-header-action ant-pro-header-dark"
              style="padding: 0px; height: 48px; line-height: 48px; width: 100%; z-index: 19;"
            >
              <div
                class="layout-header"
              >
                <div
                  class="layout-header-title"
                >
                  这是标题
                </div>
                <div
                  class="layout-header-content"
                >
                  <div
                    class="layout-header-right-content"
                  >
                    <div
                      class="layout-header-right-content-notice"
                    >
                      <img
                        alt=""
                        src="[object Object]"
                      />
                    </div>
                    <div
                      class="layout-header-right-content-avatar"
                    >
                      <div
                        class="layout-header-avatar-dropdown"
                      >
                        <div
                          class="ant-dropdown-trigger layout-header-avatar-dropdown-trigger"
                        >
                          <div
                            class="layout-header-avatar-dropdown-trigger-avatar"
                          >
                            <img
                              alt=""
                              src="[object Object]"
                            />
                          </div>
                          <div
                            class="layout-header-avatar-dropdown-trigger-name"
                          >
                            admin
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </header>
            <main
              class="ant-layout-content ant-pro-basicLayout-content ant-pro-basicLayout-has-header"
            >
              <div />
              <div
                class="home"
              >
                This is Home Page
                <div>
                  Skip Login
                </div>
                <h1>
                  This is User info
                </h1>
                <div>
                  name: test
                </div>
                <div>
                  id: test
                </div>
                <div>
                  LoginStatus: 
                  <button>
                    Logout
                  </button>
                </div>
              </div>
              <div />
              <div />
              <div>
                <div />
              </div>
            </main>
            <div
              class="layout-footer"
            >
              This is footer
            </div>
          </div>
        </section>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
